packages/services/src/view-service/address-by-index.ts

```diff
@@ -1,14 +1,13 @@
 import type { Impl } from '.';
-import { servicesCtx } from '../ctx/prax';
 
 import { getAddressByIndex } from '@penumbra-zone/wasm/src/keys';
+import { fvkCtx } from '../ctx/full-viewing-key';
 
-export const addressByIndex: Impl['addressByIndex'] = async (req, ctx) => {
-  const services = ctx.values.get(servicesCtx);
-  const {
-    viewServer: { fullViewingKey },
-  } = await services.getWalletServices();
-
+export const addressByIndex: Impl['addressByIndex'] = (req, ctx) => {
+  const fullViewingKey = ctx.values.get(fvkCtx);
+  if (!fullViewingKey) {
+    throw new Error('Cannot access full viewing key');
```

Comment:
```suggestion
    throw new ConnectError('Cannot access full viewing key', Code.Unauthenticated);
```

and the same at many similar locations